seqr/views/apis/report_api.py

```diff
@@ -416,21 +416,57 @@ def _add_row(row, family_id, row_type):
     airtable_rows = {table: [] for table in AIRTABLE_TABLE_COLUMNS.keys()}
     experiment_lookup_rows = []
     experiment_ids_by_participant = {}
+    missing_participant_ids = []
+    missing_airtable = []
+    missing_airtable_data_types = defaultdict(list)
+    missing_seqr_data_types = defaultdict(list)
     for participant in participant_rows:
         phenotype_rows += _parse_participant_phenotype_rows(participant)
         analyte = {k: participant.pop(k) for k in [SMID_FIELD, *ANALYTE_TABLE_COLUMNS[2:]]}
         analyte['participant_id'] = participant['participant_id']
 
         if not participant[PARTICIPANT_ID_FIELD]:
+            missing_participant_ids.append(participant['participant_id'])
             continue
 
-        airtable_metadata = airtable_metadata_by_participant.get(participant.pop(PARTICIPANT_ID_FIELD)) or {}
-        data_types = grouped_data_type_individuals[participant['participant_id']]
+        airtable_participant_id = participant.pop(PARTICIPANT_ID_FIELD)
+        airtable_metadata = airtable_metadata_by_participant.get(airtable_participant_id)
+        if not airtable_metadata:
+            missing_airtable.append(airtable_participant_id)
+            continue
+
+        seqr_data_types = set(grouped_data_type_individuals[participant['participant_id']].keys())
+        airtable_data_types = {dt.upper() for dt in GREGOR_DATA_TYPES if dt.upper() in airtable_metadata}
+        for data_type in seqr_data_types - airtable_data_types:
+            missing_airtable_data_types[data_type].append(airtable_participant_id)
+        for data_type in airtable_data_types - seqr_data_types:
+            missing_seqr_data_types[data_type].append(airtable_participant_id)
         _parse_participant_airtable_rows(
-            analyte, airtable_metadata, data_types, experiment_ids_by_participant,
+            analyte, airtable_metadata, seqr_data_types.intersection(airtable_data_types), experiment_ids_by_participant,
             analyte_rows, airtable_rows, experiment_lookup_rows,
         )
 
+    errors = []
+    if missing_participant_ids:
+        errors.append(
+            f'The following participants are missing {PARTICIPANT_ID_FIELD} for the airtable Sample: '
+            f'{", ".join(sorted(missing_participant_ids))}'
+        )
+    if missing_airtable:
+        errors.append(
+            f'The following entries are missing airtable metadata: '
+            f'{", ".join(sorted(missing_airtable))}'
+        )
+    warnings = [
+        f'The following entries are missing {data_type} airtable data: {", ".join(participants)}'
+        for data_type, participants in sorted(missing_airtable_data_types.items())
```

Comment:
are entries missing airtable data types also omitted / should this warning reflect that?

Reply:
so in that case where the data is not in airtable there is no data to be omitted. The data is in airtable so if its not there we aren't omitting anything